### ISSUE TITLE
skip and limit not working properly with aggregate functions?

### ISSUE THREAD

**Issue by nileshbhagwat** [ujjwalguptaofficial/JsStore#175]
# skip and limit not working properly with aggregate functions?
example: https://ujjwalguptaofficial.github.io/idbstudio/?db=Demo

- Here 'Customer' store is having total 93 records 
```
select({
    from: "Customers"
});
```

- skip and limit works fine in this scenario
```
select({
    from: "Customers",
    skip: 0,
    limit: 20
}); // returns first 20 records (1-20) ✔
```
```
select({
    from: "Customers",
    skip: 20,
    limit: 20
}); // returns next 20 records (21-40) ✔
```

- Issue: However its not working for following scenario with aggregate function
```
select({
    from: "Customers",
    aggregate: {
        count: ['customerName']
    },
    groupBy: "country"
}); // returns total 22 records ✔ 
```
```
select({
    from: "Customers",
    aggregate: {
        count: ['customerName']
    },
    groupBy: "country",
    skip: 0,
    limit: 10
}); // should return first 10 out of 22 records but returning 7 records ❌
```
```
select({
    from: "Customers",
    aggregate: {
        count: ['customerName']
    },
    groupBy: "country",
    skip: 10,
    limit: 10
}); // should return next 10 records out of 22 but returning 8 records ❌
```

**Comment by ujjwalguptaofficial** [comment]
Thanks for great explanation @nileshbhagwat , will look into issue asap.

**Comment by ujjwalguptaofficial** [comment]
@nileshbhagwat issue is resolved in new version - 3.10.1. I have also updated idbstudio with latest version, so query can be also tested on url or by  locally installing idbstudio. 

Let me know if there is any issue.
